src/datadog/tracer.cpp

```diff
@@ -175,11 +177,33 @@ Expected<Span> Tracer::extract_span(const DictReader& reader,
       return error->with_prefix(
           extraction_error_prefix(style, audited_reader.entries_found));
     }
-    extracted_contexts.push_back(std::move(*data));
-    extracted_contexts.back().headers_examined = audited_reader.entries_found;
+
+    if (!first_style_with_trace_id && (*data).trace_id.has_value()) {
```

Comment:
```suggestion
    if (!first_style_with_trace_id && data->trace_id.has_value()) {
```

We usually use the `->`